docs/usage/INVENTORY_SYNC.md

```diff
@@ -104,22 +91,88 @@ An example of use can be found [here](https://github.com/commercetools/commercet
+failed, processed categories and the processing time of the sync in different time units and in a
+human readable format.
+````java
+final InventorySyncStatistics stats = syncStatisticsStage.toCompletebleFuture().join();
+stats.getReportMessage(); 
+/*"Summary: 25 inventory entries were processed in total (9 created, 5 updated, 2 failed to sync)."*/
+````
 
-- `warningCallBack`
-a callback that is called whenever an event occurs during the sync process that represents a warning.
+__Note__ The statistics object contains the processing time of the last batch only. This is due to two reasons:
+ 1. The sync processing time should not take into account the time between supplying batches to the sync. 
+ 2. It is not known by the sync which batch is going to be the last one supplied.
 
-- `beforeUpdateCallback`
-a filter function which can be applied on a generated list of update actions. It allows the user to intercept inventory 
-entry **_update_** actions just before they are sent to CTP API.
 
-- `beforeCreateCallback`
-a filter function which can be applied on an inventoryEntry draft before a request to create it on CTP is issued. It allows the 
-user to intercept inventoryEntry **_create_** requests to modify the draft before the create request is sent to CTP API.
+More examples of how to use the sync [here](/src/integration-test/java/com/commercetools/sync/integration/inventories/InventorySyncIT.java).
+
+### Build all update actions
+
+A utility method provided by the library to compare an InventoryEntry with a new InventoryEntryDraft and results in a list of InventoryEntry
+ update actions. 
+```java
+List<UpdateAction<InventoryEntry>> updateActions = InventorySyncUtils.buildActions(oldEntry, newEntry, inventorySyncOptions);
+```
 
-<!-- TODO Update above options with links to tests. Tests should be written when inventory sync could actually use them (when custom update actions would use them).  -->
+Examples of its usage can be found in the tests 
+[here](/src/test/java/com/commercetools/sync/inventories/utils/InventorySyncUtilsTest.java).
 
-## Under the hood
+### Build particular update action(s)
+
+Utility methods provided by the library to compare the specific fields of an InventoryEntry and a new InventoryEntryDraft, and in turn builds
+ the update action. One example is the `buildChangeQuantityAction` which compares quantities:
+  
+````java
+Optional<UpdateAction<InventoryEntry>> updateAction = buildChangeQuantityAction(oldEntry, newEntry);
+````
 
-The tool matches categories by their `sku` and `supplyChannel` key. Based on that inventories are created or updated.
-Currently, the tool does not support inventory deletion.
+## Caveats
+
+1. The tool matches inventories by their `sku` and `supplyChannel` key. Inventories are either created or updated. 
+Currently the tool does not support inventory deletion.
+2. The sync library is not meant to be executed in a parallel fashion. For example:
```

Comment:
Can it be generalized as we will have to repeat almost same code block on every resource? Maybe as pseudo-code or coded once fully for product sync and then just linked from other resources.

Reply:
e6f1d6a